models/caseStudy/casestudy.js

```diff
@@ -0,0 +1,175 @@
+
+const mongoose = require('mongoose');
+
+const caseStudySchema = mongoose.Schema({
+  company_logo: {
+    type: Buffer,
+    required: false
+  },
+  main_title: {
+    type: String,
+    required: true
+  },
+  main_description: {
+    type: String,
+    required: true
+  },
+  main_company_image: {
+    type: Buffer,
+    required: false
+  },
+  company_name: {
+    type: String,
+    required: true
+  },
+  company_location: {
+    type: String,
+    required: false,
+  },
+  company_industry: {
+    type: String,
+    required: false,
+  },
+  company_ipo: {
+    type: String,
+    required: false,
+  },
+  company_website: {
+    type: String,
+    required: false,
+  },
+  company_employee_number: {
+    type: String,
+    required: false,
+  },
+  company_established_year: {
+    type: String,
+    required: false,
+  },
+  company_speciality: {
+    type: String,
+    required: false,
+  },
+  company_stage: {
+    type: String,
+    required: false,
+  },
+  company_type: {
+    type: String,
+    required: false,
+  },
+  company_personal_name: {
+    type: String,
+    required: true,
+  },
+  company_personal_role: {
+    type: String,
+    required: true,
+  },
+  company_personal_image: {
+    type: Buffer,
+    required: false,
+  },
+  company_personal_quote: {
+    type: String,
+    required: false
+  },
+  context_text: {
+    type: String,
+    required: false
+  },
+  problem_text: {
+    type: String,
+    required: false
+  },
+  solution_text: {
+    type: String,
+    required: false
+  },
+  results_text: {
+    type: String,
+    required: false
+  },
+  language: {
+    type: String,
+    required: true
+  }
+});
+
+caseStudySchema.statics.getAll = function (callback) {
+  CaseStudy.find({}, (err, case_study) => {
+    if (err) {
+      return callback('bad_request');
+    }
+    return callback(undefined, case_study);
+  })
+}
+
+caseStudySchema.statics.getSingle = function (id, callback) {
```

Comment:
Şunun ismini findCaseStudyById yap, direk id ile bulma. mongoose.Types.ObjectId(id.toString) kullan. Ama öncesinda valdiator ile id valid mi kontrol et. Örnek için herhangi bir modele bakabilirsin. Hepsinde id ile bulma fonksiyonu var. Orayı direk örnek al.